pkg/analysis/passes/checksum/checksum.go

```diff
@@ -0,0 +1,155 @@
+package checksum
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/readme"
+	"github.com/grafana/plugin-validator/pkg/logme"
+)
+
+var (
+	checksumInvalid     = &analysis.Rule{Name: "checksum-invalid", Severity: analysis.Error}
+	checksumFormatError = &analysis.Rule{Name: "checksum-format-error", Severity: analysis.Error}
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "checksum",
+	Requires: []*analysis.Analyzer{readme.Analyzer},
+	Run:      run,
+	Rules:    []*analysis.Rule{checksumInvalid},
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+
+	checksum := pass.CheckParams.Checksum
+
+	// skip if no provided
+	if checksum == "" || len(checksum) == 0 {
+		return nil, nil
+	}
+
+	checksum, err := resolveCheckSum(strings.TrimSpace(checksum))
+	if err != nil {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			checksumFormatError,
+			fmt.Sprintf("%s: %s", err.Error(), pass.CheckParams.Checksum),
+			"Make sure you provide a valid checksum as a string or a valid URL to a checksum. Only MD5 and SHA1 checksums are supported.",
+		)
+		return nil, nil
+	}
+
+	checkSumType := ""
+
+	// determine if it is md5 or sha1
+	if len(checksum) == 32 {
+		checkSumType = "md5"
+	} else if len(checksum) == 40 {
+		checkSumType = "sha1"
+	} else {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			checksumFormatError,
+			fmt.Sprintf("Invalid checksum format: %s", checksum),
+			"Make sure you provide a valid checksum as a string or a valid URL to a checksum. Only MD5 and SHA1 checksums are supported.",
+		)
+		return nil, nil
+	}
+	isError := false
+
+	switch checkSumType {
+	case "md5":
+		if checksum != pass.CheckParams.ArchiveCalculatedMD5 {
+			isError = true
+		}
+	case "sha1":
+		if checksum != pass.CheckParams.ArchiveCalculatedSHA1 {
+			isError = true
+		}
+	}
```

Comment:
Nit: Maybe make the check lower case? Something like:

```suggestion
	switch checkSumType {
	case "md5":
		if strings.ToLower(checksum) != strings.ToLower(pass.CheckParams.ArchiveCalculatedMD5) {
			isError = true
		}
	case "sha1":
		if strings.ToLower(checksum) != strings.ToLower(pass.CheckParams.ArchiveCalculatedSHA1) {
			isError = true
		}
	}
```

Reply:
great catch!